sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -467,10 +494,148 @@ The `node` class provides the {crs}[common reference semantics].
 [source,c++]
 ----
 namespace sycl::ext::oneapi::experimental {
-  class node {};
+  class node {
+    template <int Dimensions>
+    void update_nd_range(nd_range<Dimensions> executionRange);
+  };
 }
 ----
 
+Table {counter: tableNumber}. Member functions of the `node` class.
+[cols="2a,a"]
+|===
+|Member Function|Description
+
+|
+[source,c++]
+----
+template <int Dimensions>
+void update_nd_range(nd_range<Dimensions> executionRange);
+----
+| Updates the ND-Range for this node with a new value. This new value will not
+affect any executable graphs this node is part of until it is passed to the
+executable graph's update function.
+See <<executable-graph-update, Executable Graph Update>> for more information
+about updating node parameters.
+
+Parameters:
+
+* `executionRange` - The new value for the ND-Range.
+
+Exceptions:
+
+* Throws with error code `nd_range` if `executionRange` is an invalid ND-Range
+  for this node.
```

Comment:
After some offline discussion I believe we do need to restrict changing the number of dimensions to be portable across backends which don't do everything in 3D anyway (aka OpenCL). It could also be advised in the spec that if you did wish to change the number of dimensions it can be done if you provide a 3 dimensional range to begin with and just manually adjust dimensions to 1 when updating.

We also discussed that updating a kernel which was created with an `nd_range` with a new `range` value should probably also be forbidden, and vice-versa. I don't think we will have any restrictions outside of those so I'll update the errors here to reflect this.